clash-prelude/src/Clash/Explicit/Reset.hs

```diff
@@ -180,23 +184,96 @@ resetSynchronizer clk rst = rstOut
 -- See: https://github.com/clash-lang/clash-compiler/pull/2511
 {-# CLASH_OPAQUE resetSynchronizer #-} -- Give reset synchronizer its own HDL file
 
+resetGlitchFilter, unsafeResetGlitchFilter
+  :: forall dom glitchlessPeriod n
+   . ( HasCallStack
+     , KnownDomain dom
+     , glitchlessPeriod ~ (n + 1) )
```

Comment:
Why do we do this instead of `1 <= glitchlessPeriod` ?

Reply:
I don't know. @martijnbastiaan ?

Reply:
I think it used to be expressed in terms of `Vec` functions - and `Foldable (Vec n a)` imposes this constraint.

Reply:
The constraint on Foldable got changed to `1 <= n` in version 0.99 while the `resetGlitchFilter` was introduced in version 1.4.

Reply:
Ah misread. Dunno in that case. Do we still need this constraint at all?

Reply:
No, let's drop it. While it makes no sense to filter for a minimum of 0 cycles, it will actually achieve that with the current implementation (unless I'm being really dense), so let's not add unneeded constraints.

Reply:
Hmmm no, we'd need to test it to see if it works because I am apparently really dense. For instance, `maxBound :: Index 0` is `XException`. And what is `Eq (Index 0)`? There's no concrete value for `Index 0`, it's not even a singleton, it's an empty set.